### ISSUE TITLE
DragRuler not working with Foundry 12

### ISSUE THREAD

**Issue by Haditsch** [manuelVo/foundryvtt-drag-ruler#326]
When attempting to drag a token, a duplicate shows up and no tokens can be moved afterwards

**Comment by Cempres** [comment]
@manuelVo are there any plans on updating the module, as I'm not updating to v12 without this module :) Thanks for all the great work you do!

**Comment by apoapostolov** [comment]
Same here. Probably not much but @manuelVo I will add a 20$ bounty on this.

**Comment by CacciaMorta** [comment]
Same problem. The new version works only after disabling this module.

**Comment by morgan613** [comment]
Elevation Ruler works for foundry v12 and does all the drag ruler things! If I shouldn't say this here, feel free to delete comment. I figured it is okay because this is largely a volunteer based community.

**Comment by apoapostolov** [comment]
@morgan613 does it bring pathfinding? if not, Drag Ruler is a must-have for me on V12.

**Comment by morgan613** [comment]
it does!

**Comment by apoapostolov** [comment]
For some reason couldn't get it work. But still, it is nice to have competition.

I am still hoping and standing to my country @manuelVo want to see Drag Ruler still kick in for V12.

**Comment by morgan613** [comment]
@apoapostolov you couldn't get the pathfinding to work? Here is a screenshot of the UI toggle button I was referring to that is created when you enable elevation ruler. If it still isn't working feel free to email me at morgandavies613@gmail.com since it would probably be good to move the conversation elsewhere.

![Foundry Virtual Tabletop 6_4_2024 10_29_41 AM](https://github.com/manuelVo/foundryvtt-drag-ruler/assets/59232964/7cf0d978-4b47-4e22-adcf-7aebaa303fe1)

**Comment by manuelVo** [comment]
Since requests are flowing in steadily, it's possibly a good idea to state this publicly: I plan to port Drag Ruler to v12 in the coming weeks.

**Comment by iLLHunter** [comment]
fantastic news. thank you

**Comment by nitmpez** [comment]
Its a quick hack, but drag ruler atleasts works if you modify the below code in src/ruler.js line 108 from:
```js
 if (d.x === this.destination.x && d.y === this.destination.y) return;
 ```
 to:
```js
  if (d.x === destination.x && d.y === destination.y) return;
```

**Comment by Ginkgo85** [comment]
> Since requests are flowing in steadily, it's possibly a good idea to state this publicly: I plan to port Drag Ruler to v12 in the coming weeks.

When is the update coming?

**Comment by manuelVo** [comment]
> When is the update coming?

Whenever I find a time slot where I have sufficient time and motivation remaining to work on this. Work is pretty stressful right now and my evenings are short (and weekends fully booked). Unfortunately this means I cannot give you an estimation for when this will happen.

**Comment by manuelVo** [comment]
Thanks to @TPNils work (and some fixes by @n0q) a v12 Version is now released.
